tests/runtime/out_loki.c

```diff
@@ -783,6 +784,180 @@ void flb_test_float_value()
     flb_destroy(ctx);
 }
 
+static void cb_check_structured_metadata_value(void *ctx, int ffd,
+                                 int res_ret, void *res_data, size_t res_size,
+                                 void *data)
+{
+    char *p;
+    flb_sds_t out_js = res_data;
+    if (!TEST_CHECK(out_js != NULL)) {
+        TEST_MSG("out_js is NULL");
+        return;
+    }
+
+    char *index_line = (char *) data;
+
+    p = strstr(out_js, index_line);
+    if (!TEST_CHECK(p != NULL)) {
+      TEST_MSG("Expecting %s but Given:%s", index_line, out_js);
+    }
+
+    flb_sds_destroy(out_js);
+}
+
+#define JSON_MAP "[12345678, {\"log\": \"This is an interesting log message!\", " \
+    "\"map1\": {\"key1\": \"value1\", \"key2\": \"value2\", \"key_nested_object_1\": " \
+    "{\"sub_key1\": \"sub_value1\", \"sub_key2\": false}}, \"map2\": {\"key4\": " \
+    "\"value1\", \"key5\": false}, \"map3\": {\"key1\": \"map3_value1\", \"key2\": " \
+    "\"map3_value2\"}}]"
+void flb_test_structured_metadata_map_params(char *remove_keys,
```

Comment:
There are quite a few different variations on this test, so I have created this function which takes various parameters, and then called it for each actual test (below)